_summaries/Rae2019compressive_1.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: summary
+title: Summary
+paper: Rae2019compressive_1
+# Please fill out info below
+author: zayne-sprague
+score: 9
+---
+
+## What's the big idea
+
+- By compressing memories, transformers are able to model long range sequences more efficiently and effectively, 
+outperforming state-of-the-art benchmarks on various sequence based tasks.
+  
+## What's Wrong With Attention
+
+- Attention is expensive in both computation time and memory
+  
+- It's also not a great model of how memory really works (most do not remember every detail of every second of their lives, yet that is the transformer model)
+
+- Current architecture is forced to move on and limit themselves to a fixed size window of attention
+
+![animation of transformerXL](rae2019compressive_1_custom_b.gif)
+
+
+## How Can We Improve Attention
+
+- Not all memories deserve the same amount of memory or detail (think of the resolution of short-term vs long-term memories)
+  
+- We can reduce the dimensions (and thus the resolution of the representation) for a subset of memories in the sequence
+
+    - Compressed memories are now coarser and have a rough analogy to long-term memories
+    - UnCompressed memories often contain more important information and have a rough analogy to short-term memories
+
+  
+## How Does This Work (How Is It Realized)
+
+1.) You pass a sequence $$S_i = x_t, ..., x_n$$ through the network
+
+2.) As the model progresses to the next sequence $$S_i$$ is moved into memory of past activation (we can still attend to it)
+
+3.) Eventually, the memory gets to large so we evict the oldest memory
+
+4.) Instead of deleting the oldest memory, we create a mapping to compress it and store it in a stack
+
+![ How compression works ](rae2019compressive_1_custom_a.png)
+
+
+## How do we compress an activation aka "memory"
+
+- The main idea is to use a mapping from uncompressed to compressed: $$f_c: R^{n_s \ x \ d} \rightarrow R^{[\frac{n_s}{c}] \ x \ d}$$
```

Comment:
It'd be helpful to define what each hyperparameter means -- in particular n_s and c, which directly control the size of the Transformer and the compression rate.

Reply:
Totally right, I added a brief bullet point list explaining each.  Good call